entity/entity.h

```diff
@@ -18,10 +19,9 @@ class RenderPass;
 
 class Entity {
  public:
-  /// All pipeline blend mode presets assume that both the source (fragment
-  /// output) and destination (first color attachment) have colors with
-  /// premultiplied alpha.
-  enum class BlendMode {
+  /// Basic blends are blends which can be performed using the standard pipeline
+  /// blend equations.
+  enum class BasicBlendMode {
```

Comment:
Not a huge fan of `Basic` and `Advanced` as those terms are ambiguous to the user of the API. Now I see why you called these `PipelineBlendMode`s when you first added the patch. How about just combining the various blend modes (ones that need a fragment shader as well as the ones that can be done via the pipeline equations) into a single enum. Internally, you can disambiguate. I feel like that would be a better API UX. Skia does something similar too I believe.

Reply:
Done. Yeah I wasn't stoked about having separate enums either. I did it mostly for type safety -- maybe combining them is the lesser of two evils. The "advanced" terminology came from an internal Skia enum for blend equations. I may refactor this a bit more later in a future PR and move the pipeline blend parameter resolution out of `ContentContextOptions`. We'll probably want the freedom to do funky blend mode tricks in texture filters.

I tried to make it as safe as possible by adding validations and static asserts where appropriate, but let me know if you have other tricks you normally like to do in this situation.